### PR TITLE
Add MDISynopticMap source

### PR DESCRIPTION
**Pull request by dstansby**
Similar to HMI, MDI synoptic maps need a handful of meta fixes.

### CODE REVIEW

File: sunpy/map/sources/soho.py [hunk 1]
```diff
@@ -204,4 +212,29 @@ def measurement(self):
...
+    """
+    SOHO MDI synoptic magnetogram Map.
+
+    See the docstring of `MDIMap` for information on the MDI instrument.
+    """
+    def __init__(self, data, header, **kwargs):
+        # FITS doesn't like "Degree" as a unit
+        if header['cunit1'] == 'Degree':
+            header['cunit1'] = 'deg'
+        # Sine Latitude is not a valid unit - see Thompson 2005, section 5.5
```
Comment (dstansby) [inline sunpy/map/sources/soho.py:228]:
```suggestion
        # Sine Latitude is not a valid unit - see Thompson 2006, section 5.5
```

File: sunpy/map/sources/tests/test_mdi_source.py [hunk 2]
```diff
@@ -23,6 +26,94 @@ def mdi():
     return Map(fitspath)
 
 
+@pytest.fixture
+def mdi_synoptic():
+    # sample MDI map
+    # Please do not edit this header, it is a 1:1 copy of a MDI map header
+    header = dedent("""
+        # HDU 0 in mdi.synoptic_mr_small_96m.2099.data.fits:
```
Comment (dpshelio) [inline sunpy/map/sources/tests/test_mdi_source.py:34]: Would be better to keep this in a separate txt file?

File: sunpy/map/sources/soho.py [hunk 3]
```diff
@@ -173,9 +173,17 @@ def __init__(self, data, header, **kwargs):
         vmin = np.nanmin(self.data)
         vmax = np.nanmax(self.data)
         if abs(vmin) > abs(vmax):
-            self.plot_settings['norm'] = colors.Normalize(-vmin, vmin)
+            self.plot_settings['norm'] = colors.Normalize(-abs(vmin), abs(vmin))
         else:
-            self.plot_settings['norm'] = colors.Normalize(-vmax, vmax)
+            self.plot_settings['norm'] = colors.Normalize(-abs(vmax), abs(vmax))
```
Comment (dpshelio) [inline sunpy/map/sources/soho.py:178]: I would suggest a DRY-er approach:
```python
threshold = max([abs(vmin), abs(vmax)])
self.plot_settings['norm'] = colors.Normalize(-threshold, threshold)

```

File: sunpy/map/sources/tests/test_mdi_source.py (hunk 2 above)
Comment (dstansby) [reply sunpy/map/sources/tests/test_mdi_source.py:33]: I think others are just in the test code files? Happy to move it to a text file, but not sure where the best place to put that would be.

Comment (nabobalis) [reply sunpy/map/sources/tests/test_mdi_source.py:33]: I'm happy to keep it here over a text file.

**Comment by nabobalis** [comment]
Will need rebasing so the CI passes.

**Comment by nabobalis** [comment]
Will need rebasing again due to the WCS warning now in master.

**Comment by nabobalis** [comment]
Looks like some legit test failures:
```
>       assert contour.observer == simple_map.observer_coordinate.frame
E       assert <HeliographicStonyhurst Coordinate (obstime=2020-07-17T09:19:10.627): (lon, lat, radius) in (deg, deg, m)\n    (0., 0., 1.)> == <HeliographicStonyhurst Coordinate (obstime=2020-07-17T09:19:10.627): (lon, lat, radius) in (deg, deg, m)\n    (0., 0., 1.)>
E         -<HeliographicStonyhurst Coordinate (obstime=2020-07-17T09:19:10.627): (lon, lat, radius) in (deg, deg, m)\n
E         -    (0., 0., 1.)>
E         +<HeliographicStonyhurst Coordinate (obstime=2020-07-17T09:19:10.627): (lon, lat, radius) in (deg, deg, m)\n
E         +    (0., 0., 1.)>
```

**Comment by nabobalis** [comment]
Failure is due to me. I fixed it here: https://github.com/sunpy/sunpy/pull/4383

**Pull request merged**